docs/test-prep.md

```diff
@@ -0,0 +1,406 @@
+
+        <!-- Fill field "Price" with "500.00" -->
+        <fillField selector="{{AdminProductFormSection.productPrice}}" userInput="{{_defaultProduct.price}}" stepKey="fillPriceField" />
+
+        <!-- Fill field "Quantity" with "100" -->
+        <fillField selector="{{AdminProductFormSection.productQuantity}}" userInput="{{_defaultProduct.quantity}}" stepKey="fillQtyField" />
+
+        <!-- Fill field "Weight" with "100" -->
+        <fillField selector="{{AdminProductFormSection.productWeight}}" userInput="{{_defaultProduct.weight}}" stepKey="fillWeightField" />
+
+       ...
+    </test>
+</tests>
+```
+
+One of the reasons that we abstract things is so that they are more flexible and reusable. In this case, we can leverage this flexibility and use an existing data file. For this scenario, we are using [this file](https://raw.githubusercontent.com/magento-pangolin/magento2/MageTestFest/app/code/Magento/Catalog/Test/Mftf/Data/ProductData.xml).
+
+Data entities are important because this is where a `suffix` or `prefix` can be defined. This ensures that data values can be unique for every test run.
+
+Notice that the `<entity>` name is `_defaultProduct` as referenced above. Within this entity is the `name` value.
+
+```xml
+<entities xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:DataGenerator/etc/dataProfileSchema.xsd">
+    <entity name="_defaultProduct" type="product">
+        <data key="sku" unique="suffix">testSku</data>
+        <data key="type_id">simple</data>
+        <data key="attribute_set_id">4</data>
+        <data key="visibility">4</data>
+   ---> <data key="name" unique="suffix">testProductName</data>
+        <data key="price">123.00</data>
+        <data key="urlKey" unique="suffix">testurlkey</data>
+        <data key="status">1</data>
+        <data key="quantity">100</data>
+        <data key="weight">1</data>
+        <requiredEntity type="product_extension_attribute">EavStockItem</requiredEntity>
+        <requiredEntity type="custom_attribute_array">CustomAttributeCategoryIds</requiredEntity>
+    </entity>
+```
+
+The `unique="suffix"` attribute appends a random numeric string to the end of the actual data string. This ensures that unique values are used for each test run.
+
+XXXX for difficult data needs, see https://devdocs.magento.com/mftf/docs/data.html
+
+## Convert actions to action groups
+
+Action groups are sets of steps that are run together. Action groups are designed to break up multiple individual steps into logical groups. For example: Logging into the admin panel requires ensuring the login form exists, filling in 2 form fields and clicking the Submit button. These can be bundled into a single, reusable "Admin Login" action group that can be applied to any other test. This leverages existing code and prevents duplication of effort. We recommend that all steps in a test be within an action group.
```

Comment:
in 2 form fields -> in two form fields

Reply:
the Submit button -> the **Submit** button
Here and in all other cases, use bold formatting for UI element names.

Reply:
Got it.